utils/serde/src/main/java/io/apicurio/registry/utils/serde/AbstractKafkaSerializer.java

```diff
@@ -74,6 +91,7 @@ public AbstractKafkaSerializer(
             T schema = toSchema(data);
             String artifactId = getArtifactIdStrategy().artifactId(topic, isKey(), schema);
             long id = getGlobalIdStrategy().findId(getClient(), artifactId, artifactType(), schema);
+            schema = getCache().getSchema(id); // use registry's schema!
```

Comment:
@alesj @EricWittmann why it's required to fetch the just created schema? this is causing the tests to fail in the streams storage variant. Because of the storage being asynchronous , fetching the schema here can fail (and so it's doing in the tests)

Reply:
> why it's required to fetch the just created schema?

Since you can end-up with wrong schema -- data is actually invalid aka not schema compliant.
And it would all still work w/o problems -- on the producer side.
It would fail on the consumer side, but I guess you want to catch this before -- so not to waist network and Kafka brokers, where it could be prevented.

Reply:
so what do you suggest? active wait? (to ensure the artifact gets created in the registry) or "manually" load the cache with the schema we already have in memory? or what else is possible?

Reply:
It seems like we need to manually load the cache with the schema, but only in the case where the global ID strategy stored the schema in the registry.  This is the only use-case where the fetch will fail when it should not (due to the async nature of the Streams impl).  @alesj and @famartinrh agreed?

Reply:
agree on loading the cache under that conditions, now is up to how are we going to implement that :)

Reply:
> This is the only use-case where the fetch will fail when it should not (due to the async nature of the Streams impl)

Ah yes ... yeah, that can be async ... and not much that can be done about it, other than retry ?

Reply:
Or ... actually, in some cases you already know that it's gonna create new globalId, so you could pre-populate the cache ... so no need for another globalId->schema lookup ... but that depends on the strategy ...

Reply:
My suggestion is that we add a retry here **for now** so that all the tests pass.  And then we add a GitHub issue to properly fix the issue by pre-caching the schema in the case where the strategy actually uploads it to the registry.

Any objections?

Reply:
Hearing no objections, I have taken this approach.  Note that I also had to update the CheckPeriod test so that it worked against async storages.